crates/trigger-http/src/lib.rs

```diff
@@ -391,6 +386,40 @@ impl HttpTrigger {
         });
     }
 
+    async fn instrumented_service_fn(
+        self: Arc<Self>,
+        addr: SocketAddr,
+        request: Request<Incoming>,
+    ) -> Result<Response<HyperOutgoingBody>> {
+        let span = tracing::info_span!(
```

Comment:
Could this be moved into a `create_span(addr, request)` type function (and then we could, if desired, retain the existing structure, but with a call to that function before the `self_.handle`)?  It feels like this is mixing the concerns of "defining the instrumentation" and "handling the request".